include/sdbus-c++/IConnection.h

```diff
@@ -133,6 +135,25 @@ namespace sdbus {
          */
         virtual std::string getUniqueName() const = 0;
 
+        /*!
+         * @brief Attach the bus connection to an sd-event event loop
+         *
+         * @throws sdbus::Error in case of failure
+         */
+        virtual void attachEventLoop(sd_event *ev, int priority = 0) = 0;
```

Comment:
Suggestion. Because there are already `enterEventLoop()` and `enterEventLoopAsync()` functions defined, and they are not compatible with these new event loop related calls, it might be helpful to be more explicit with the naming. E.g. `attachSdEventLoop()`, `detachSdEventLoop()`, etc.

Another thing that came to mind, was the possibility of using the `sd-event` loop as the built-in event loop to make the integration consistent. However, I am not aware of any hooks `sd-event` provides to notify an application when the event loop is about to start and end processing. Thus, there would be no way to know when to lock `sdbusMutex_` and preserve the thread-safety of the bus state. So, unless there is something I am missing, that is not possible with the current `libsystemd` API.

Reply:
I will change those names.

Regarding integration of sd-event as the event loop I think it should be doable by not using sd_event_loop. Take a look at https://www.freedesktop.org/software/systemd/man/sd_event_run.html you could go a step or two lower level. Or maybe I'm not fully understanding how it should work.

Reply:
Done